plugins/autoslab.cpp

```diff
@@ -228,7 +210,14 @@ static void checkslabs(color_ostream &out)
     for (auto ghost : ghosts)
     {
         // Only create a job is the map has no existing jobs for that historical figure or no existing engraved slabs
-        if (histToJob.count(ghost->hist_figure_id) == 0 && !engravedSlabItemExists(ghost, engravedSlabs))
+        if (histToJob.count(ghost->hist_figure_id) == 0 &&
+            !std::any_of(engravedSlabs.begin(),
+                        engravedSlabs.end(),
+                        [&ghost](df::item *slab){
+                            auto slabst = virtual_cast<df::item_slabst>(slab);
```

Comment:
this caught my eye here since it's usually dangerous to virtual_cast without checking for NULL. it should always work here, but you wouldn't have to convert if `engravedSlabs` were a `std::vector<df::item_slabst>`, and you can cut down on your scanning *and* get the correct type if you scan `world->items.other.SLAB` instead of `world->items.all`